x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -0,0 +1,287 @@
+  };
+
+  public async componentDidMount() {
+    const indexPattern = await getAPMIndexPattern();
+    this.setState({ hasIndexPattern: !!indexPattern });
+  }
+
+  public componentDidUpdate(prevProps: FlyoutProps) {
+    if (
+      prevProps.urlParams.transactionType !==
+      this.props.urlParams.transactionType
+    ) {
+      this.setState({
+        selectedTransactionType: this.props.urlParams.transactionType
+      });
+    }
+  }
+
+  public createJob = async () => {
+    this.setState({ isLoading: true });
+    try {
+      const { serviceName, transactionType } = this.props.urlParams;
+      if (serviceName) {
+        const res = await startMLJob({ serviceName, transactionType });
+        const didSucceed = res.datafeeds[0].success && res.jobs[0].success;
+        if (!didSucceed) {
+          throw new Error('Creating ML job failed');
+        }
+        this.addSuccessToast();
+      }
+    } catch (e) {
+      this.addErrorToast();
+    }
+
+    this.setState({ isLoading: false });
+    this.props.onClose();
+  };
+
+  public addErrorToast = () => {
+    const { location, urlParams } = this.props;
+    const { serviceName = 'unknown', transactionType } = urlParams;
+
+    if (!serviceName) {
+      return;
+    }
+
+    toastNotifications.addWarning({
+      title: 'Job already exists',
+      text: (
+        <p>
+          There&apos;s already a job running for anomaly detection on{' '}
+          {serviceName} ({transactionType}).{' '}
+          <ViewMLJob
+            serviceName={serviceName}
+            transactionType={transactionType}
+            location={location}
+          >
+            View existing job
+          </ViewMLJob>
+        </p>
+      )
+    });
+  };
+
+  public addSuccessToast = () => {
+    const { location, urlParams } = this.props;
+    const { serviceName = 'unknown', transactionType } = urlParams;
+
+    toastNotifications.addSuccess({
+      title: 'Job successfully created',
+      text: (
+        <p>
+          The analysis is now running for {serviceName} ({transactionType}
+          ). It might take a while before results are added to the response
+          times graph.{' '}
+          <ViewMLJob
+            serviceName={serviceName}
+            transactionType={transactionType}
+            location={location}
+          >
+            View job
+          </ViewMLJob>
+        </p>
+      )
+    });
+  };
+
+  public render() {
+    const { isOpen, onClose, urlParams } = this.props;
+    const { serviceName, transactionType } = urlParams;
+    const { isLoading, hasIndexPattern, selectedTransactionType } = this.state;
+
+    if (!isOpen || !serviceName) {
+      return null;
+    }
+
+    return (
+      <MLJobsRequest
+        serviceName={serviceName}
+        render={({ data, status }) => {
+          if (status === 'LOADING') {
+            return null;
+          }
+
+          const hasMLJob = Boolean(
+            data.jobs.find(
```

Comment:
Nit: `data.jobs.some` would probably be more appropriate.

Reply:
Good call